src/renderer/components/ft-video-player/ft-video-player.js

```diff
@@ -18,6 +18,16 @@ import { calculateColorLuminance, colors } from '../../helpers/colors'
 import { pathExists } from '../../helpers/filesystem'
 import { getPicturesPath, showSaveDialog, showToast } from '../../helpers/utils'
 
+// YouTube now throttles if you use the `Range` header for the DASH formats, instead of the range query parameter
+// videojs-http-streaming calls this hook everytime it makes a request,
+// so we can use it to convert the Range header into the range query parameter for the streaming URLs
+videojs.Vhs.xhr.beforeRequest = (options) => {
+  if (options.headers?.Range && new URL(options.uri).hostname.endsWith('.googlevideo.com')) {
```

Comment:
I'm guessing we check if it's .googlevideo.com because we dont need this if the video is proxied through invidious?

Reply:
The Invidious proxy accepts both, but by the looks of it, it only sends YouTube the Range header, so they should probably fix that. https://github.com/iv-org/invidious/blob/master/src/invidious/routes/video_playback.cr

Reply:
Actually looks like the Invidious proxy dies after a few requests and just sends HTTP 416 status codes if you use the range query parameter.

Reply:
![invidious-proxy-error](https://user-images.githubusercontent.com/48293849/221958933-2322384a-7052-4d4f-836e-6009de917a42.png)